test/e2e/set/scylladbmanagertask/scylladbmanagertask_globalmanager_object_storage.go

```diff
@@ -33,7 +36,454 @@ import (
+		sc, err = f.ScyllaAdminClient().ScyllaV1alpha1().ScyllaDBClusters(f.Namespace()).Create(ctx, sc, metav1.CreateOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = utils.RegisterCollectionOfRemoteScyllaDBClusterNamespaces(ctx, sc, rkcClusterMap)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the ScyllaDBCluster %q roll out (RV=%s)", sc.Name, sc.ResourceVersion)
+		waitSDBCRolloutCtx, cancelSDBCRollout := utils.ContextForMultiDatacenterScyllaDBClusterRollout(ctx, sc)
+		defer cancelSDBCRollout()
+		sc, err = controllerhelpers.WaitForScyllaDBClusterState(waitSDBCRolloutCtx, f.ScyllaAdminClient().ScyllaV1alpha1().ScyllaDBClusters(sc.Namespace), sc.Name, controllerhelpers.WaitForStateOptions{}, utils.IsScyllaDBClusterRolledOut)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		scylladbclusterverification.Verify(ctx, sc, rkcClusterMap)
+		err = scylladbclusterverification.WaitForFullQuorum(ctx, rkcClusterMap, sc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Inserting data into the ScyllaDBCluster")
+		hostsByDC := make(map[string][]string, len(sc.Spec.Datacenters))
+		for _, dc := range sc.Spec.Datacenters {
+			remoteClusterClient, ok := rkcClusterMap[dc.RemoteKubernetesClusterName]
+			o.Expect(ok).To(o.BeTrue(), "Remote Kubernetes cluster %q should be present in the map", dc.RemoteKubernetesClusterName)
+
+			dcStatus, _, ok := oslices.Find(sc.Status.Datacenters, func(dcStatus scyllav1alpha1.ScyllaDBClusterDatacenterStatus) bool {
+				return dc.Name == dcStatus.Name
+			})
+			o.Expect(ok).To(o.BeTrue(), "Datacenter %q should be present in ScyllaDBCluster status", dc.Name)
+			o.Expect(dcStatus.RemoteNamespaceName).ToNot(o.BeNil(), "Remote namespace name for datacenter %q should not be nil", dc.Name)
+
+			sdcName := naming.ScyllaDBDatacenterName(sc, &dc)
+			sdc, err := remoteClusterClient.ScyllaAdminClient().ScyllaV1alpha1().ScyllaDBDatacenters(*dcStatus.RemoteNamespaceName).Get(ctx, sdcName, metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			sourceHosts, err := utilsv1alpha1.GetBroadcastRPCAddresses(ctx, remoteClusterClient.KubeAdminClient().CoreV1(), sdc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+			hostsByDC[dc.Name] = sourceHosts
+		}
+
+		di := verification.InsertAndVerifyCQLDataByDC(ctx, hostsByDC)
+		defer di.Close()
+
+		framework.By("Registering ScyllaDBCluster with the global ScyllaDB Manager instance")
+		smClusterRegistration := &scyllav1alpha1.ScyllaDBManagerClusterRegistration{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "smcr-",
+				Namespace:    ns.Name,
+			},
+			Spec: scyllav1alpha1.ScyllaDBManagerClusterRegistrationSpec{
+				ScyllaDBClusterRef: scyllav1alpha1.LocalScyllaDBReference{
+					Kind: scyllav1alpha1.ScyllaDBClusterGVK.Kind,
+					Name: sc.Name,
+				},
+			},
+		}
+		smClusterRegistration, err = nsClient.ScyllaClient().ScyllaV1alpha1().ScyllaDBManagerClusterRegistrations(ns.Name).Create(ctx, smClusterRegistration, metav1.CreateOptions{})
```

Comment:
hint: smcr is an internal resource, it shouldn't be created directly but by labelling the ScyllaDBCluster. This part should be pretty much identical to the existing test for ScyllaDBDatacenter